tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+        };
+        methodRequest({ method: 'HEAD', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should return AccessForbidden for OPTIONS request on bucket without ' +
+    'CORSConfiguration', done => {
+        const headers = {
+            'Origin': 'http://www.allowedwebsite.com',
+            'Access-Control-Request-Method': 'GET',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+        headersResponse: null }, done);
+    });
+
+    describe('allow PUT, POST, DELETE, GET methods and allow only ' +
+    'one origin', () => {
+        const corsParams = {
+            Bucket: bucket,
+            CORSConfiguration: {
+                CORSRules: [
+                    {
+                        AllowedMethods: [
+                            'PUT', 'POST', 'DELETE', 'GET',
+                        ],
+                        AllowedOrigins: [
+                            'http://www.allowedwebsite.com',
+                        ],
+                    },
+                ],
+            },
+        };
+        beforeEach(done => {
+            s3.putBucketCors(corsParams, done);
+        });
+
+        afterEach(done => {
+            s3.deleteBucketCors({ Bucket: bucket }, done);
+        });
+
+        it('send OPTIONS request with allowed origin', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'GET',
+            };
+            const headersResponse = {
+                'access-control-allow-origin': 'http://www.allowedwebsite.com',
+                'access-control-allow-methods': 'PUT, POST, DELETE, GET',
+                'access-control-allow-credentials': 'true',
+                'vary': 'Origin, Access-Control-Request-Headers, ' +
+                    'Access-Control-Request-Method',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 200,
+            headersResponse }, done);
+        });
+        it('send OPTIONS request with not allowed request headers', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'GET',
+                'Access-Control-Request-Headers': 'Origin, Accept, ' +
+                'Content-Type',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+        it('send OPTIONS request with not allowed origin', done => {
+            const headers = {
+                'Origin': 'http://www.forbiddenwebsite.com',
+                'Access-Control-Request-Method': 'GET',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+        it('send OPTIONS request with listed method "POST"', done => {
```

Comment:
the post, put and delete tests look like they are the same.  perhaps this is a good place for a limited forEach test to just loop through those 3 cases.